apps/cdd-backend/src/test-utils/mocks.ts

```diff
@@ -111,3 +111,11 @@ export class MockPolymesh {
     },
   };
 }
+
+export const slackAppMock = {
+  client: {
+    chat: {
+      postMessage: jest.fn().mockResolvedValue(true),
+    },
+  },
+};
```

Comment:
We have the golevelup auto mocker. I would reccomend something like this in the test case:

```suggestion
let mockSlack: DeepMocked<SlackApp>

beforeEach(() => {
  mockSlack = createMock<SlackApp>()
})
```

Then you will have a fresh instance and can use `.toHaveBeenCalledWith` instead of `toHaveBeenNthCalledWith`

Reply:
If you refactor to a service you can inject the module with this in the `providers`

```ts
  { provide: SlackApp, useValue: mockSlack }
```